crates/subspace-archiving/tests/integration/piece_reconstruction.rs

```diff
@@ -114,3 +116,105 @@ fn piece_reconstruction_works() {
         blake2b_256_254_hash(missing_piece.as_ref())
     );
 }
+
+#[test]
+fn piece_reconstructor_creation_fails() {
+    let kzg = Kzg::random(PIECES_IN_SEGMENT).unwrap();
+
+    let reconstructor = PiecesReconstructor::new(10, 1, kzg.clone());
+
+    assert!(reconstructor.is_err());
+
+    if let Err(error) = reconstructor {
+        assert_eq!(error, ReconstructorInstantiationError::SegmentSizeTooSmall);
+    }
+
+    let reconstructor = PiecesReconstructor::new(10, 12, kzg);
+
+    assert!(reconstructor.is_err());
+
+    if let Err(error) = reconstructor {
+        assert_eq!(
+            error,
+            ReconstructorInstantiationError::SegmentSizesNotMultipleOfRecordSize
+        );
+    }
```

Comment:
You can test both in one line with https://doc.rust-lang.org/stable/std/assert_matches/index.html
Archiver tests already use it.